tools/fasm2frame.py

```diff
@@ -16,25 +44,49 @@ def get_database(segtype):
 
     segbitsdb[segtype] = {}
 
+    def process(l):
+        l = l.strip()
+
+        # CLBLM_L.SLICEL_X1.ALUT.INIT[10] 29_14
+        parts = line.split()
+        name = parts[0]
+        bit_vals = parts[1:]
+
+        # Assumption
+        # only 1 bit => non-enumerated value
+        if len(bit_vals) == 1:
+            seg_word_column, word_bit_n, isset = parsebit(bit_vals[0])
+            if not isset:
+                raise Exception(
+                    "Expect single bit DB entries to be set, got %s" % l)
+            segbitsdb[segtype][name] = {None: (seg_word_column, word_bit_n)}
+        else:
+            # An enumerated value
+            # Split the base name and selected key
+            m = re.match(r'(.+)[.](.+)', name)
+            name = m.group(1)
+            key = m.group(2)
+            # May or may not be the first key encountered
+            m = segbitsdb[segtype].get(name, {})
+            segbitsdb[segtype][name] = m
+            l = []
+            m[key] = l
+            #print("Add %s.%s" % (name, key))
+            for bit_val in bit_vals:
```

Comment:
This loop feels ripe for being a list comprehensions:

m[key] = [parsebit(x) for x in bit_vals]

No need to unpack the results of parsebit() just to repack them into a tuple again.

Reply:
Changed in PR